src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -1650,6 +1650,18 @@ static void vthread_monitor_waited_event(JavaThread *current, ObjectWaiter* node
   current->frame_anchor()->clear();
 }
 
+void ObjectMonitor::post_monitor_wait(JavaThread *current, jlong millis) {
```

Comment:
Not sure this is worthwhile factoring out for a few reasons:
1. the actual code is only 2 lines
2.  the comment only makes sense when a path that will park follows this code - which is now only one of the three cases
3.  It is confusing to see this method and the static `post_monitor_wait_event`

I think we can just inline the code where needed and place the comment after L1725 (as I don't think it even applies to the virtual thread case prior to that).

Reply:
Thanks. Good suggestion, I've updated the fix, mach5 tiers are in progress.

Reply:
Mach5 tiers 1-6 are good.